vhdbuilder/packer/configure-windows-vhd.ps1

```diff
@@ -321,8 +321,22 @@ function Update-WindowsFeatures {
 }
 
 function Update-Registry {
+
+    param (
+        $containerRuntime
+    )
+
+    # if multple LB policies are included for same endpoint then HNS hangs.
+    # this fix forces an error
     Write-Host "Enable a HNS fix in 2021-2C"
     reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\hns\State /v HNSControlFlag /t REG_DWORD /d 1
```

Comment:
Could you help to change all 'reg add' in this file to `Set-ItemProperty`?
Example: Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\hns\State" -Name HNSControlFlag -Value 1 -Type DWORD

Reply:
Could you please explain the benefit of the change, thanks?

Reply:
This is from aks-engine. I think that this uses PowerShell commands but not an external tool 'reg'. PowerShell commands should return/raise an error in the same format.

Reply:
So are we going to use powersehell commands to replace all external tools, like curl.

Reply:
Not curl. curl is used for automatically retry.

Reply:
It seems like that 'reg add ' is a cmd command

Reply:
So a powershell command should be better.